### PR TITLE
Draft NLP metric

### PR DESCRIPTION
**Pull request by richard-rogers**
Adds
* `VectorComponent` for holding matrices and vectors
* `SvdMetricConfig`, `SvdMetric`, and `UpdatableSvdMetric` for vector-space embeddings
* `NlpConfig`, `NlpMetric`, and `log_nlp` for NLP logging

### CODE REVIEW

File: python/whylogs/extras/example.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    _preprocessifier,
+)
+
+# from nltk.tokenize import word_tokenize
+
+# import nltk
+# nltk.download('stopwords')
+# nltk.download('inaugural')
+
+# inverted index weighting utility functions
```
Comment (jamie256) [inline python/whylogs/extras/example.py]: remove code commented out?

File: python/whylogs/extras/example.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        '"',
+        "'",
+        " ",
+        "?",
+        "[",
+        "]",
+        ".]",
+        "' ",
+        '" ',
+        "?",
```
Comment (jamie256) [inline python/whylogs/extras/example.py]: are lines 65 and 71 different?

**Review by jamie256 — CHANGES_REQUESTED** [review]
Can you clean this up a bit and get the CI green? Then I'll take a more thorough look.

Can you make the example.py an example notebook or test?

File: python/examples/experimental/NLP.ipynb [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+      "source": [
+        "## Installing whylogs\n",
+        "\n",
+        "If you haven't already, install whylogs: "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "pip install whylogs-1.1.7-py3-none-any.whl"
```
Comment (jamie256) [inline python/examples/experimental/NLP.ipynb]: remove the pip install referencing whylogs wheel, you have a working pip install in next cell.

File: python/examples/experimental/NLP.ipynb [hunk 4]
```diff
@@ -0,0 +1,567 @@
...
+        "        \".]\",\n",
+        "        \"' \",\n",
+        "        '\" ',\n",
+        "        \"? \",\n",
+        "        \"-\",\n",
+        "        \"- \",\n",
+        "        \"/\",\n",
+        "        '?\"',\n",
+        "        \"...\",\n",
+        "        \"\",\n",
```
Comment (jamie256) [inline python/examples/experimental/NLP.ipynb:194]: Is this empty string as a stop word?

**Review by jamie256 — DISMISSED** [review]
Some minor comments, but looks good to try out in experimental mode.

File: python/examples/experimental/NLP.ipynb (hunk 4 above)
Comment (richard-rogers) [reply python/examples/experimental/NLP.ipynb:194]: Yes, NLTK's tokenizer produces some odd tokens. I augmented the stop word list with the ones I saw to clean it up a bit.

File: python/whylogs/experimental/core/metrics/nlp_metric.py [hunk 5]
```diff
@@ -0,0 +1,454 @@
+from copy import deepcopy
```
Comment (richard-rogers) [inline python/whylogs/experimental/core/metrics/nlp_metric.py:1]: Should probably move this to `python/whylogs/experimental/extras`

**Pull request closed without merging**